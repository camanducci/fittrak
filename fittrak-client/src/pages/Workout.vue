<template>
  <v-container>
    <WorkoutDataContainer :workout-id="$route.params.workoutId">
      <WorkoutHeader 
        slot="header" 
        slot-scope="data" 
        :workout="data.workout" />
      <ExerciseActions 
        slot="actions" 
        slot-scope="data" 
        :workout="data.workout" />
      <ExerciseList 
        slot="list" 
        slot-scope="data" 
        :exercises="data.exercises" />
    </WorkoutDataContainer>
  </v-container>
</template>

<script>
import WorkoutDataContainer from "@/components/workouts/WorkoutDataContainer";
import WorkoutHeader from "@/components/workouts/WorkoutHeader";
import ExerciseActions from "@/components/exercises/ExerciseActions";
import ExerciseList from "@/components/exercises/ExerciseList";

export default {
  name: "WorkoutDetail",

  components: {
    WorkoutDataContainer,
    WorkoutHeader,
    ExerciseActions,
    ExerciseList
  }
};
</script>

<style>
</style>
