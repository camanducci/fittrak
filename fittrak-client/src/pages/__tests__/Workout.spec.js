import { shallowMount } from "@vue/test-utils";

import Workout from "@/pages/Workout";

describe("Workout", () => {
  test("Workout page renders", () => {
    const app = shallowMount(Workout, {
      mocks: {
        $apollo: {
          loading: false
        }
      }
    });
    expect(app.html()).toMatchSnapshot();
  });
});
