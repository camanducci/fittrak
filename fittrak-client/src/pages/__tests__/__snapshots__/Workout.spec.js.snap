// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workout Workout page renders 1`] = `
<v-container-stub tag="div" row="" wrap="">
  <v-layout-stub tag="div" row="" wrap="">
    <workoutheader-stub workout="[object Object]"></workoutheader-stub>
    <exerciseactions-stub workout="[object Object]"></exerciseactions-stub>
    <exerciselist-stub exercises=""></exerciselist-stub>
  </v-layout-stub>
</v-container-stub>
`;
